### DISCUSSION TITLE
Shouldn't we use a ternary here?

### DISCUSSION THREAD

**Discussion by rmccu** [jongpie/NebulaLogger#378] [Q&A] [answered]
https://github.com/jongpie/NebulaLogger/blob/3b04138cdb22211796870ef49d8b3773465641e9/nebula-logger/core/main/log-management/classes/LogEntryEventHandler.cls#L82

It seems like using a ternary to set logOwnerId would make more sense here.

Also, rather than compare logEntryEvent.UserLicenseName__c to guestUserLicenseName using ==, shouldn't we use the .equals() method?

**Comment by jamessimone** [comment]
It's not very idiomatic to call `.equals()` directly when doing comparisons. I'll let @jongpie respond further, but looking at the excerpt, I would imagine the `if/else` block is used because the ternary would have been likewise indented and - at that point - `if/else` is explicitly more readable.

**Comment by jongpie** [comment, marked as answer]
I agree with @jamessimone, I don't personally like using `.equals()` in this context, I think using `==` is more idiomatic.

But @rmccu as far as this particular block of code, you're looking at an out of date version - the commit you're referencing, [3b04138](https://github.com/jongpie/NebulaLogger/commit/3b04138cdb22211796870ef49d8b3773465641e9), is 4-5 months old (v4.7.1). In the latest version, [v4.8.3](https://github.com/jongpie/NebulaLogger/blob/1a4b9b650bc39c7b2b405f4830b4ff1ed4bce4c3/nebula-logger/core/main/log-management/classes/LogEntryEventHandler.cls#L409), the functionality for determining the log owner ID has grown, so there's now a private method that determines the correct owner ID.

**Reply by rmccu** [reply]
On the face of it, I agree. However, I have been moving away from the `==` comparison because it does not behave consistently across languages. Specifically in Apex, the `==` is a case insensitive comparison, which may not be obvious to the reader. Whereas using `equals()` or `equalsIgnoreCase()` is very explicit and obvious to the reader.
```
String q = 'teXt';
system.debug('teXt'.equals(q)); // true
system.debug('teXt' == q);      // true
system.debug('text'.equals(q)); // false
system.debug('text' == q);      // true
```
